content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/future-work.md

```diff
@@ -0,0 +1,55 @@
+
+### GPU calculation support
+
+The terrain system is built on top of the gradient components, which perform all of their calculations CPU-side. However, the types of calculations being performed are ideal for GPU-side calculation. An overhaul or replacement of the gradient system that supports GPU-side calculations could provide orders of magnitude improvements to the terrain system. The primary caveats are that the data is still needed CPU-side to feed to physics, generalized raycasts, and other systems, and that it should be possible to still use the terrain system on devices with limited or no GPU capabilities (low-end phones, headless servers, etc).
+
+## Additional features
+
+This category of improvements should be possible to incrementally add into the system without fundamentally altering anything about how the system currently works.
+
+### Triangle split direction
+
+The terrain currently always triangulates each quad in the same uniform direction. This could be replaced with heuristics for choosing the best direction for each quad. This could even be exposed as multiple heuristics - same direction for every quad, alternating directions per quad, best choice per quad based on vertex heights. This decision should be controlled at the terrain system level, so that the information can be provided consistently to the terrain debugger, terrain physics, terrain rendering, terrain raycasting, etc.
+
+### Improved shape to height workflows
+
+It's currently somewhat non-intuitive to create terrain heights directly from primitive shape components. The best options are to use either the Shape Falloff Gradient or the Surface Altitude Gradient, but they're both currently problematic. The Shape Falloff Gradient creates falloff based on distance from the bottom of a box, not just simply the position of the shape within a box, so it's hard to control. The Surface Altitude Gradient doesn't support falloff and the auto-refresh doesn't work. Either of these components could be improved, or a new component could be added, to make it easier to just place a shape with falloff into the world and turn it into a height gradient.
+
+### Masking / blending
+
+Add controls to the Terrain Layer Spawner and the Terrain Macro Material to allow each one to mask and blend instead of completely replacing the data. This would make it much easier to create small terrain "stamps" for things like craters.
+
+### First-class "hole" support
+
+Holes can be created right now by authoring a high-priority terrain layer spawner with no ground plane for whatever size is desired. However, it would be nice to have additional authoring controls to make it possible to put holes directly into gradient data, possibly either through a second alpha channel, or through a separate gradient on the Terrain Height Gradient List component.
```

Comment:
```suggestion
Holes can be created right now by authoring a high-priority terrain layer spawner with no ground plane for whatever size is desired. However, it will be nice to have additional authoring controls to make it possible to put holes directly into gradient data, possibly either through a second alpha channel, or through a separate gradient on the Terrain Height Gradient List component.
```